pyomo/common/fileutils.py

```diff
@@ -0,0 +1,492 @@
+#  ___________________________________________________________________________
```

Comment:
These functions look really useful - good addition. Is there an easy way for a Pyomo user to add a directory to the search path for executables / libraries, etc.? Specifically without needing to edit the PATH environment variable, I mean. There are many users who are not familiar with environment variables and maybe this would simplify the fix when Pyomo is unable to find their solver.

Reply:
Yes. `find_library()` and `find_executable()` both take a `pathlist` argument where users can specify / override the list of search paths.  The `Executable` manager has a similar attribute that a user can set:
```python
Executable.pathlist.append(new_dir)
Executable.rehash() # causes the Executable manager to 
#     re-do the search for all known executables
```
Users can also always just *set* the path to the executable of choice:
```python
Executable('my_solver').executable = path_to_my_solver
```
and that will persist despite changes to the `pathlist` or calls to `rehash()`

Having said all of that, there is currently no global "registry" for libraries akin to the `Executable` manager.  Perhaps we should add one later?